tests/drivers/spi/spi_slave/src/main.c

```diff
@@ -0,0 +1,454 @@
+}
+
+ZTEST(spi_slave, test_only_tx)
+{
+	test_only_tx(false);
+}
+
+ZTEST(spi_slave, test_only_tx_async)
+{
+	test_only_tx(true);
+}
+
+/** Test where only master transmits and slave receives in chunks. */
+static void test_only_tx_in_chunks(bool async)
+{
+	size_t len1 = 7;
+	size_t len2 = 8;
+
+	/* MTX buffer */
+	tdata.bufs[0].buf = buf_alloc(len1 + len2, true);
+	tdata.bufs[0].len = len1 + len2;
+	tdata.sets[0].buffers = &tdata.bufs[0];
+	tdata.sets[0].count = 1;
+	tdata.mtx_set = &tdata.sets[0];
+	tdata.mrx_set = NULL;
+
+	/* STX buffer */
+	tdata.bufs[1].buf = buf_alloc(len1, false);
+	tdata.bufs[1].len = len1;
+	tdata.bufs[2].buf = buf_alloc(len2, false);
+	tdata.bufs[2].len = len2;
+	tdata.sets[1].buffers = &tdata.bufs[1];
+	tdata.sets[1].count = 2;
+	tdata.srx_set = &tdata.sets[1];
+	tdata.stx_set = NULL;
+
+	run_test(true, true, async);
+}
+
+ZTEST(spi_slave, test_only_tx_in_chunks)
+{
+	test_only_tx_in_chunks(false);
+}
+
+ZTEST(spi_slave, test_only_tx_in_chunks_async)
+{
+	test_only_tx_in_chunks(true);
+}
+
+/** Test where only slave transmits. */
+static void test_only_rx(bool async)
+{
+	size_t len = 16;
+
+	/* MTX buffer */
+	tdata.bufs[0].buf = buf_alloc(len, true);
+	tdata.bufs[0].len = len;
+	tdata.sets[0].buffers = &tdata.bufs[0];
+	tdata.sets[0].count = 1;
+	tdata.mrx_set = &tdata.sets[0];
+	tdata.mtx_set = NULL;
+
+	/* STX buffer */
+	tdata.bufs[1].buf = buf_alloc(len, false);
+	tdata.bufs[1].len = len;
+	tdata.sets[1].buffers = &tdata.bufs[1];
+	tdata.sets[1].count = 1;
+	tdata.stx_set = &tdata.sets[1];
+	tdata.srx_set = NULL;
+
+	run_test(true, true, async);
+}
+
+ZTEST(spi_slave, test_only_rx)
+{
+	test_only_rx(false);
+}
+
+ZTEST(spi_slave, test_only_rx_async)
+{
+	test_only_rx(true);
+}
+
+/** Test where only slave transmits in chunks. */
+static void test_only_rx_in_chunks(bool async)
+{
+	size_t len1 = 7;
+	size_t len2 = 9;
+
+	/* MTX buffer */
+	tdata.bufs[0].buf = buf_alloc(len1 + len2, true);
+	tdata.bufs[0].len = len1 + len2;
+	tdata.sets[0].buffers = &tdata.bufs[0];
+	tdata.sets[0].count = 1;
+	tdata.mrx_set = &tdata.sets[0];
+	tdata.mtx_set = NULL;
+
+	/* STX buffer */
+	tdata.bufs[1].buf = buf_alloc(len1, false);
+	tdata.bufs[1].len = len1;
+	tdata.bufs[2].buf = buf_alloc(len2, false);
+	tdata.bufs[2].len = len2;
+	tdata.sets[1].buffers = &tdata.bufs[1];
+	tdata.sets[1].count = 2;
+	tdata.stx_set = &tdata.sets[1];
+	tdata.srx_set = NULL;
+
+	run_test(true, true, async);
+}
+
+ZTEST(spi_slave, test_only_rx_in_chunks)
+{
+	test_only_rx_in_chunks(false);
+}
+
+ZTEST(spi_slave, test_only_rx_in_chunks_async)
+{
+	test_only_rx_in_chunks(true);
+}
+
+static void before(void *not_used)
+{
+	ARG_UNUSED(not_used);
+
+	memset(spim_buffer, 0, sizeof(spim_buffer));
+	memset(spis_buffer, 0, sizeof(spis_buffer));
+	memset(&tdata, 0, sizeof(tdata));
+	for (size_t i = 0; i < sizeof(spim_buffer); i++) {
+		spim_buffer[i] = (uint8_t)i;
+	}
+	for (size_t i = 0; i < sizeof(spis_buffer); i++) {
+		spis_buffer[i] = (uint8_t)i;
```

Comment:
It would be better to use different values in `spis_buffer` (e.g. `0x80 + i`) to ensure that the required bytes were actually transferred (i.e. buffer contents have changed) in test cases where only one side transmits.